api/utils/strings.go

```diff
@@ -98,3 +99,13 @@ func CopyStrings(in []string) []string {
 
 	return out
 }
+
+// MapToString converts a map into a string.
+func MapToString(m map[string]string) string {
+	var sb strings.Builder
+	for key, value := range m {
+		sb.WriteString(key)
+		sb.WriteString(value)
+	}
+	return sb.String()
+}
```

Comment:
Why not just maintain the separate strings? I have no idea if we already have a utility function like this one, but

```suggestion
// MapToStrings collects keys and values of a map into a []string.
func MapToStrings(m map[string]string) []string {
	s := make([]string, 0, len(m) * 2)
	for key, value := range m {
		s = append(s, key, value)
	}
	return s
}
```

and then this can be used in-place like
```go
fieldVals := append(utils.MapToStrings(a.GetAllLabels()), a.GetName(), a.GetDescription(), a.GetPublicAddr())
```

It would also avoid the strange behavior of matching `"vpro"` to `{"env": "prod"}`.

Reply:
> It would also avoid the strange behavior of matching "vpro" to {"env": "prod"}.

good point

> I have no idea if we already have a utility function like this one

We do have this: [LabelsAsString](https://github.com/gravitational/teleport/blob/2d958d69571f5c028ceb51e55ce9d01e9c520f2c/api/types/server.go#L329), but i figured b/c it did extra work that was unnecessary (ie. sorting), we create a simpler one.